### ISSUE TITLE
Parsing error for `external()` receiver

### ISSUE THREAD

**Issue by novusnota** [tact-lang/tact#220]
Steps to reproduce:
1. Make a new TS project, with `@tact-lang/compiler` as a dependency
2. Add the following code to it's main file:

```typescript
import { check, VirtualFileSystem, CheckResult } from '@tact-lang/compiler';
import { resolve, normalize, sep } from 'path';
import { existsSync, readFileSync } from 'fs';

// Helper class for creating a fs
export class OverwritableVirtualFileSystem implements VirtualFileSystem {
  root: string;
  overwrites: Map<string, Buffer> = new Map();

  constructor(root: string) {
    this.root = normalize(root);
    if (!this.root.endsWith(sep)) {
      this.root += sep;
    }
  }

  resolve(...path: string[]): string {
    return normalize(resolve(this.root, ...path));
  }

  exists(path: string): boolean {
    return existsSync(path);
  }

  readFile(path: string): Buffer {
    return this.overwrites.get(path) ?? readFileSync(path);
  }

  writeFile(path: string, content: string | Buffer): void {
    this.overwrites.set(path, typeof content === 'string' ? Buffer.from(content, 'utf-8') : content);
  }
}

const fs = new OverwritableVirtualFileSystem(process.cwd());
const checkResults: CheckResult = check({
  project: fs,
  entrypoint: 'test.tact' // path to the contract
});
if (checkResults.ok === false) {
  for (const checkResult of checkResults.messages) {
    console.log(checkResult);
  }
}
```

3. And the `test.tact` file:

```tact
contract HelloWorld {
    init() {
        // ...
    }

    get fun greeting(): String {
        return "Hello, World!";
    }

    receive() {
        dump(context());
    }

    // this is making things go awry:
    external() {
        42;
    }
}
```

4. Then, upon running this TypeScript project (can use `ts-node`, for example), the following is outputted to the console:

```shell
{
  type: 'error',
  message: "Missing semantic action for 'ReceiveFunction_externalEmpty' in operation 'resolve_declaration'.\n" +
    'Action stack (most recent call last):\n' +
    '  resolve_program > Program\n' +
    "  resolve_program_item > default action for 'ProgramItem'\n" +
    "  resolve_program_item > default action for 'Contract'\n" +
    '  resolve_program_item > Contract_simple\n' +
    "  resolve_declaration > default action for 'ContractBody'\n" +
    "  resolve_declaration > default action for 'ReceiveFunction'\n" +
    '  resolve_declaration > ReceiveFunction_externalEmpty',
  location: { file: 'src/test.tact', line: 0, column: 0, length: 0 }
}
```

Note, that without the `external()` things work smoothly.

This is observed on Tact version: 1.2.0.

**Comment by anton-trunov** [comment]
What happens if we replace `external() { 42; }` with, for instance, `external() { }` (external with the empty body)?

**Comment by novusnota** [comment]
> What happens if we replace `external() { 42; }` with, for instance, `external() { }` (external with the empty body)?

Same error

**Comment by anton-trunov** [comment]
It was not an API bug, but a parser bug and it's been apparently fixed in #341
